tests/gtest/avifallocationtest.cc

```diff
@@ -0,0 +1,177 @@
+// Copyright 2022 Google LLC. All rights reserved.
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include <array>
+#include <limits>
+#include <tuple>
+
+#include "avif/avif.h"
+#include "aviftest_helpers.h"
+#include "gtest/gtest.h"
+
+using ::testing::Combine;
+using ::testing::Values;
+using ::testing::ValuesIn;
+
+namespace libavif {
+namespace {
+
+void TestAllocation(uint32_t width, uint32_t height, uint32_t depth,
+                    avifResult expected_result) {
+  // The format of the image and which planes are allocated should not matter.
+  // Test all combinations.
+  for (avifPixelFormat format :
+       {AVIF_PIXEL_FORMAT_NONE, AVIF_PIXEL_FORMAT_YUV444,
```

Comment:
[Just a comment, no change requested] I am surprised that `avifImageAllocatePlanes()` allows `image->yuvFormat` to be `AVIF_PIXEL_FORMAT_NONE`.

Reply:
It is to store alpha only, from what I understood.